iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/io/WALInputStream.java

```diff
@@ -0,0 +1,284 @@
+      tryLoadSegment();
+    }
+  }
+
+  private void loadNextSegmentV1() throws IOException {
+    // just read raw data as input
+    if (channel.position() >= fileSize) {
+      throw new IOException("Unexpected end of file");
+    }
+    if (Objects.isNull(dataBuffer)) {
+      // read 128 KB
+      dataBuffer = ByteBuffer.allocate(128 * 1024);
+    }
+    dataBuffer.clear();
+    channel.read(dataBuffer);
+    dataBuffer.flip();
+  }
+
+  private void loadNextSegmentV2() throws IOException {
+    headerBuffer.clear();
+    if (channel.read(headerBuffer) != Integer.BYTES + 1) {
+      throw new IOException("Unexpected end of file");
+    }
+    // compressionType originalSize compressedSize
+    headerBuffer.flip();
+    CompressionType compressionType = CompressionType.deserialize(headerBuffer.get());
+    int dataBufferSize = headerBuffer.getInt();
+    if (compressionType != CompressionType.UNCOMPRESSED) {
+      compressedHeader.clear();
+      if (channel.read(compressedHeader) != Integer.BYTES) {
+        throw new IOException("Unexpected end of file");
+      }
+      compressedHeader.flip();
+      int uncompressedSize = compressedHeader.getInt();
+      dataBuffer = ByteBuffer.allocateDirect(uncompressedSize);
+      ByteBuffer compressedData = ByteBuffer.allocateDirect(dataBufferSize);
+      if (channel.read(compressedData) != dataBufferSize) {
+        throw new IOException("Unexpected end of file");
+      }
+      compressedData.flip();
+      IUnCompressor unCompressor = IUnCompressor.getUnCompressor(compressionType);
+      dataBuffer.clear();
+      unCompressor.uncompress(compressedData, dataBuffer);
+    } else {
+      dataBuffer = ByteBuffer.allocate(dataBufferSize);
+      if (channel.read(dataBuffer) != dataBufferSize) {
+        throw new IOException("Unexpected end of file");
+      }
+    }
+    dataBuffer.flip();
+  }
+
+  private void tryLoadSegment() throws IOException {
+    long originPosition = channel.position();
+    try {
+      loadNextSegmentV2();
+      version = FileVersion.V2;
+    } catch (Throwable e) {
+      // failed to load in V2 way, try in V1 way
+      logger.warn("Failed to load WAL segment in V2 way, try in V1 way", e);
+      channel.position(originPosition);
+    }
+
+    if (version == FileVersion.UNKNOWN) {
+      loadNextSegmentV1();
+      version = FileVersion.V1;
+    }
+  }
+
+  public void skipToGivenPosition(long pos) throws IOException {
+    if (version == FileVersion.V2) {
+      channel.position(WALWriter.MAGIC_STRING_BYTES);
+      ByteBuffer buffer = ByteBuffer.allocate(Byte.BYTES + Integer.BYTES);
+      long posRemain = pos;
+      int currSegmentSize = 0;
+      while (posRemain > 0) {
+        buffer.clear();
+        channel.read(buffer);
+        buffer.flip();
+        buffer.get();
+        currSegmentSize = buffer.getInt();
+        if (posRemain >= currSegmentSize) {
+          posRemain -= currSegmentSize;
+        } else {
+          break;
+        }
+      }
```

Comment:
Where is the segment skipped? I only see the header being read.

Reply:
Fix it.